CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinks.h

```diff
@@ -0,0 +1,300 @@
+			FocalPointPlasticityInternalLink(_initiator, getCellGFromConst(_initiated), _potts, _lambdaDistance, _targetDistance, _maxDistance, _initMCS) {}
+		FocalPointPlasticityInternalLink(const CellG *_initiator, const CellG *_initiated, Potts3D *_potts, float _lambdaDistance = 0.0, float _targetDistance = 0.0, float _maxDistance = 100000.0, int _initMCS = 0) :
+			FocalPointPlasticityInternalLink(getCellGFromConst(_initiator), getCellGFromConst(_initiated), _potts, _lambdaDistance, _targetDistance, _maxDistance, _initMCS) {}
+
+		const long getId0() { return initiator->id; }
+		const long getId1() { return initiated->id; }
+
+	};
+
+	class FOCALPOINTPLASTICITY_EXPORT FocalPointPlasticityAnchor : public FocalPointPlasticityLinkBase {
+
+		FocalPointPlasticityLinkType type = FocalPointPlasticityLinkType::ANCHOR;
+
+	public:
+		FocalPointPlasticityAnchor() {}
+		FocalPointPlasticityAnchor(CellG *_cell, Potts3D *_potts, FocalPointPlasticityLinkTrackerData _fppltd)
+		{
+			initiator = _cell;
+			initiated = (CellG*)(0);
+			potts = _potts;
+			fppltd = FocalPointPlasticityLinkTrackerData(_fppltd);
+			fppltd.anchor = true;
+			fppltd.anchorPoint = _fppltd.anchorPoint;
+
+			DerivedProperty<FocalPointPlasticityLinkBase, float, &FocalPointPlasticityLinkBase::getDistance> length(this);
+			DerivedProperty<FocalPointPlasticityLinkBase, float, &FocalPointPlasticityLinkBase::getTension> tension(this);
+		}
+		FocalPointPlasticityAnchor(const CellG *_cell, Potts3D *_potts, FocalPointPlasticityLinkTrackerData _fppltd) :
+			FocalPointPlasticityAnchor(getCellGFromConst(_cell), _potts, _fppltd) {}
+		FocalPointPlasticityAnchor(CellG *_cell, Potts3D *_potts, FocalPointPlasticityTrackerData _fpptd) :
+			FocalPointPlasticityAnchor(_cell, _potts, FocalPointPlasticityLinkTrackerData(_fpptd)) {}
+		FocalPointPlasticityAnchor(const CellG *_cell, Potts3D *_potts, FocalPointPlasticityTrackerData _fpptd) :
+			FocalPointPlasticityAnchor(getCellGFromConst(_cell), _potts, _fpptd) {}
+		FocalPointPlasticityAnchor(CellG *_cell, Potts3D *_potts, float _lambdaDistance = 0.0, float _targetDistance = 0.0, float _maxDistance = 100000.0, int _initMCS = 0, std::vector<float> _anchorPoint = std::vector<float>(3, 0.0)) :
+			FocalPointPlasticityAnchor(_cell, _potts, FocalPointPlasticityLinkTrackerData(_lambdaDistance, _targetDistance, _maxDistance, _initMCS)) {}
+		FocalPointPlasticityAnchor(const CellG *_cell, Potts3D *_potts, float _lambdaDistance = 0.0, float _targetDistance = 0.0, float _maxDistance = 100000.0, int _initMCS = 0, std::vector<float> _anchorPoint = std::vector<float>(3, 0.0)) :
+			FocalPointPlasticityAnchor(getCellGFromConst(_cell), _potts, _lambdaDistance, _targetDistance, _maxDistance, _initMCS, _anchorPoint) {}
+
+		const long getId0() { return initiator->id; }
+		const long getId1() { return fppltd.anchorId; }
+
+		// Get anchor point
+		std::vector<float> getAnchorPoint() { return fppltd.anchorPoint; }
```

Comment:
So really, the only thing that warrans existence of those 3 classes of links is anchor point. But concenptually link is a linik "regular" link hooks to a moving target anchor link is hooked to stationary target. This is the only difference and I feel like we should have just one class for those 3 types of links External, interna, anchor. My motivation is code simplicity although I can see how you can make argument to keep 3 classes

Reply:
I think we we add getAnchorPoint() setAnchorPoint getAnchorId to  "unified" links class and throw exception when user tries to set e.g. anchorPoint on a non-anchor link this woudl be a decent compromise and the code woudl be much simpler. What do you think?

Reply:
See my commentary above. In general, I'm with you on simplicity. Handling the subtle differences between link types seemed better suited for their class definitions, rather than everywhere they are integrated. The generic architecture also seemed like a better approach for supporting future dev of link definitions and new link types. I'm also not particularly fond of putting members on class definitions that are only relevant for particular interpretations of the class instance (e.g., putting getAnchorPoint() on an external link type). It seems like this might cause more problems in the future for us and the user than the added value of simpler class code.